### ISSUE TITLE
Documentation search broken with Sphinx 3.0

### ISSUE THREAD

**Issue by mdickinson** [enthought/traits#1046]
After building the documentation with Sphinx 3.0 and the latest version of the Enthought Sphinx Theme, I get broken search result links from Sphinx.

Search itself works fine, but if I click on one of the search result links it takes me to a malformed url like the following:
```
file:///Users/mdickinson/Enthought/ETS/traits/docs/build/traits_user_manual/internalsundefined?highlight=hastraits
```

Replacing "internalsundefined" with "internals.html" gives the correct URL.

**Comment by mdickinson** [comment]
It's fine with Sphinx 2.4.4. This smells like a missing configuration option somewhere.

**Comment by mdickinson** [comment]
It's also fine with themes other than the Enthought Sphinx Theme, which points to a bug with that theme.

**Comment by mdickinson** [comment]
Partial diagnosis: the `undefined` string in the URL is coming from `DOCUMENTATION_OPTIONS.LINK_SUFFIX` in `searchtools.js`. If I hand-edit the js file to replace that with `.html`, everything  works as expected.

There's a `_static/documentation_options.js` file in the built documentation which defines `DOCUMENTATION_OPTIONS.LINK_SUFFIX`, but I suspect that that file is no longer being used by the base theme.

**Comment by mdickinson** [comment]
There's also a `DOCUMENTATION_OPTIONS` block at the start of `search.html` that doesn't include `LINK_SUFFIX`; hand-editing `LINK_SUFFIX` in also makes the search work as expected.

**Comment by mdickinson** [comment]
Fix in enthought/enthought-sphinx-theme#12.

**Comment by mdickinson** [comment]
This is fixed by the enthought-sphinx-theme 0.6.2 release. Waiting on availability of enthought-sphinx-theme in the buildsystem before closing this.

xref: enthought/buildsystem#3096

**Comment by mdickinson** [comment]
enthought-sphinx-theme is now available in the buildsystem, and docs built with the latest Sphinx and theme from the buildsystem are working well. Closing.
